ui/index.js

```diff
@@ -177,6 +179,24 @@ async function startApp(metamaskState, backgroundConnection, opts) {
     );
   }
 
+  // This block autoswitches chains based on the last chain used
+  // for a given dapp, when there are no pending confimrations
+  // This allows the user to be connected on one chain
+  // for one dapp, and automatically change for another
+  const state = store.getState();
+  const networkIdToSwitchTo = getNetworkToAutomaticallySwitchTo(state);
+
+  if (networkIdToSwitchTo) {
+    await store.dispatch(
+      actions.automaticallySwitchNetwork(
+        networkIdToSwitchTo,
+        getOriginOfCurrentTab(state),
+      ),
+    );
+  } else if (process.env.MULTICHAIN) {
```

Comment:
Nit: Maybe for readability, we could put all of this in a `if (process.env.MULTICHAIN) {` block? Otherwise this gives the impression that we might be switching in non-multichain builds, but never clearing the switch notice.

That might speed up initialization a little for the non-multichain case as well.

Reply:
Done!